File: server/supply-chain-domain/supplier-service/src/main/java/com/metawebthree/supplier/domain/entity/Supplier.java
package com.metawebthree.supplier.domain.entity;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Supplier {
    private Long id;
    private String supplierCode;
    private String supplierName;
    private String supplierType;
    private String businessLicense;
    private String taxId;
    private String province;
    private String city;
    private String district;
    private String address;
    private String contact;
    private String phone;
    private String email;
    private String status;
    private BigDecimal creditLimit;
    private String paymentTerms;
    private String category;
    private Integer score;
    private String level;
    private String assessmentLevel;
    private String name;
    private String contactPerson;
    private String contactPhone;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public boolean isActive() {
        return "ACTIVE".equals(status);
    }

    public void activate() {
        this.status = "ACTIVE";
    }

    public void deactivate() {
        this.status = "INACTIVE";
    }
}

File: apps/backstage-admin/src/apis/supplier.ts
import http from '@/utils/http'

export interface Supplier {
  id?: number
  supplierCode: string
  supplierName: string
  name?: string
  supplierType?: string
  province?: string
  city?: string
  address?: string
  contact?: string
  contactPerson?: string
  contactPhone?: string
  phone?: string
  email?: string
  status?: string
  creditLimit?: number
  paymentTerms?: string
  category?: string
  score?: number
  level?: string
  assessmentLevel?: string
  createdAt?: string
  updatedAt?: string
}

export interface SupplierQueryParam {
  pageNum?: number
  pageSize?: number
  status?: string
  category?: string
  supplierCode?: string
  supplierName?: string
}

export function getSupplierListAPI(params: SupplierQueryParam) {
  return http<Supplier[]>({
    url: '/api/supplier/suppliers',
    method: 'get',
    params: params,
  })
}

export function getSupplierByIdAPI(id: number) {
  return http<Supplier>({
    url: '/api/supplier/suppliers/' + id,
    method: 'get',
  })
}

export function getSupplierByCodeAPI(code: string) {
  return http<Supplier>({
    url: '/api/supplier/suppliers/code/' + code,
    method: 'get',
  })
}

export function createSupplierAPI(data: Supplier) {
  return http<Supplier>({
    url: '/api/supplier/suppliers',
    method: 'post',
    data: data,
  })
}

export function updateSupplierAPI(id: number, data: Supplier) {
  return http<Supplier>({
    url: '/api/supplier/suppliers/' + id,
    method: 'put',
    data: data,
  })
}

export function updateSupplierAssessmentAPI(id: number, level: string) {
  return http<Supplier>({
    url: '/api/supplier/suppliers/' + id + '/assessment',
    method: 'put',
    params: { level },
  })
}